server/schedulers/hot_region.go

```diff
@@ -1212,3 +1213,85 @@ func getRegionStatKind(rwTy rwType, dim int) statistics.RegionStatKind {
 	}
 	return 0
 }
+
+func (bs *balanceSolver) preferPriority() string {
+	priorities := bs.sche.conf.WritePriorities
+	if bs.rwTy == read {
+		priorities = bs.sche.conf.ReadPriorities
+	}
+	return priorities[0]
+}
+
+func calRank(greatDecRatio, minorDecRatio float64, priorities []string,
+	dimHotList [statistics.DimLen]bool, dimDecRatioList [statistics.DimLen]float64) int64 {
+	better := 1
+	noWorse := 0
+	worse := -1
+	priorityLen := len(priorities)
+	originValue := 0
+	for i, priority := range priorities {
+		dimIndex := priorityToDim(priority)
+		dimHot := dimHotList[dimIndex]
+		dimDecRatio := dimDecRatioList[dimIndex]
+		x := 0
+		switch {
+		case dimHot && dimDecRatio < greatDecRatio:
+			x = better
+		case dimDecRatio < minorDecRatio:
+			x = noWorse
+		case dimDecRatio >= minorDecRatio:
+			x = worse
+		}
+		originValue = originValue + x*getValue(priorityLen-i-1)
```

Comment:
in master, `byte is no worse and key is better` is better than ` byte is better, whatever key`, but in this pr, it will be worse.